libs/shared/flagd-core/src/lib/flagd-core.ts

```diff
@@ -91,35 +92,47 @@ export class FlagdCore {
       }
     }
 
-    const defaultVariant = flag.defaultVariant;
-
-    let resolvedVariant;
+    let variant;
     let reason;
 
-    if (flag.targetingString == undefined) {
-      resolvedVariant = flag.variants.get(defaultVariant);
+    if (!flag.targeting) {
+      variant = flag.defaultVariant;
       reason = StandardResolutionReasons.STATIC;
     } else {
-      // todo - targeting evaluation
-      // till targeting is handled, return the static result
-      resolvedVariant = flag.variants.get(defaultVariant);
-      reason = StandardResolutionReasons.STATIC;
+      let targetingResolution;
+      try {
+        targetingResolution = this._targeting.applyTargeting(flagKey, flag.targeting, evalCtx);
+      } catch (e) {
+        console.error(`Error evaluating targeting rule for flag ${flagKey}, falling back to default`, e);
```

Comment:
Can we follow the safe logger pattern used in the SDK? That could be handled in a follow up task.

Reply:
this is done for the evaluator contract